### PR TITLE
feat(scheduler): add scheduler tests that do not go through normal api tests

### PR DESCRIPTION
**Pull request by helgi**
This expands the scheduler test suite and can run outside of the API setup. Brings is simple testing per resource type.

Also fixes a few mock and scheduler discrepencies that were discovered in the testing

### CODE REVIEW

**Comment by helgi** [comment]
There are more tests to write here, just PRing this for some visual codecov.io response and to get visibility on the work

File: rootfs/scheduler/__init__.py [hunk 1]
```diff
@@ -1841,6 +1839,10 @@ def _build_deployment_manifest(self, namespace, name, image, entrypoint, command
         return manifest
 
     def _scale_deployment(self, namespace, name, image, entrypoint, command, **kwargs):
+        """
+        A conveniance wrapper around Deployment update that does a little bit of introspection
```
Comment (arschles) [inline rootfs/scheduler/__init__.py]: s/conveniance/convenience

File: rootfs/scheduler/utils.py [hunk 2]
```diff
@@ -1,4 +1,65 @@
...
+        'sailfish', 'sandwich', 'sculptor', 'seashore', 'seedcake', 'skylight', 'stickpin',
+        'tabletop', 'tailbone', 'teamwork', 'teaspoon', 'tinkerer', 'traverse', 'turbojet',
+        'umbrella', 'underdog', 'undertow', 'unicycle', 'universe', 'uptowner', 'utensils',
+        'vacation', 'vagabond', 'valkyrie', 'variable', 'villager', 'vineyard', 'vocalist',
+        'waggoner', 'waxworks', 'waterbed', 'wayfarer', 'whitecap', 'windmill', 'woodshed',
+        'yachting', 'yardbird', 'yardwork', 'yearbook', 'yearling', 'yeomanry', 'yodeling',
+        'zaniness', 'zeppelin', 'ziggurat', 'zillions', 'zirconia', 'zoologer', 'zucchini',
+    ]
+    return "{}-{}".format(
+        random.choice(adjectives), random.choice(nouns))
```
Comment (arschles) [inline rootfs/scheduler/utils.py]: was this moved from somewhere else?

Comment (helgi) [inline rootfs/scheduler/utils.py]: yes, from api.utils, it's a copy of it - trying to keep scheduler unaware of api

**Comment by arschles** [comment]
Test code looks good, and I'm ok with falling short of the coverage target, since this goes toward that goal

**Pull request merged**